src/lib.rs

```diff
@@ -604,6 +604,7 @@ use ice_::IceAgentEvent;
 pub use ice_::{Candidate, CandidateKind, IceConnectionState};
 
 /// Low level ICE access.
+#[cfg_attr(not(feature = "ice-agent"), doc(hidden))]
```

Comment:
So this just hides the module from the docs unless the feature is activated? Fine with me but I'd be curious to understand the thinking!

Reply:
Yeah, it does.

For new users coming to  str0m, I want the Main API to be clear and obvious. We're already breaking with convention by not using the RTCPeerConnection API, so I want a new user's first impressions to lead them straight to where they need to be.

This is why I'm very particular which structs/enums you see in the root – they should all be relevant and things you probably have to interact with as a normal user. `IceCreds` is right there on the boundary – it's possible to get the value from `RtcConfig`, but I don't see a regular user needing it, hence I moved it.

The ice agent is an internal concern to such a user (you never need it and there's no way to insert or interact with it through config or the `Rtc` instance).

Reply:
Thank you for explaining, keeping the root module tidy is definitely something that I consider important. What I don't really understand is why we need a feature-flag for that.

Would it be better to just use `doc(hidden)` unconditionally? I can use the `ice `module without activating the feature, so why would I activate it? To look at the docs locally? If you don't build with all features in docs.rs, then advanced users won't be able to discover the API. If you do build with all features, then the API is documented again, potentially confusing new users.

FWIW: I consider building in docs.rs with all features to be good practice. Together with `doc_auto_cfg`, that explains to users, which features they must activate to get access to certain modules. See `rust-libp2p` for some examples:

- Building with all features in `docs.rs`: https://github.com/libp2p/rust-libp2p/blob/4260702f749d45fb53ea7edd2fb0a728afb83488/libp2p/Cargo.toml#L150-L155
- Activation of `doc_auto_cfg`: See https://github.com/libp2p/rust-libp2p/blob/4260702f749d45fb53ea7edd2fb0a728afb83488/libp2p/src/lib.rs#L33
- Rendered docs.rs: https://docs.rs/libp2p/latest/libp2p/autonat/index.html

I don't think `str0m` needs it at this point because there aren't any features yet (worth documenting). But as we introduce the `crypto` module, this might be something to consider.

Reply:
For a while there I was trying to see if we can avoid compiling the rest of str0m when you only want the ice stuff. What I did was.

1. Refactor `Rtc` into `rtc.rs` and `RtcConfig` into `config.rs`
2. Add another feature `webrtc` that is on default and added to most things in `lib.rs`

Sadly it became more of a mess than had hoped for – `#[cfg(feature = "webrtc")]` in a bit too many places. When I also realised the dependency between the modules ice -> parser had to be sorted somehow, I lost interest. 

At this point maybe `#[doc(hidden)]` would be cleaner, true.

Reply:
> At this point maybe `#[doc(hidden)]` would be cleaner, true.

I'll send a follow-up!